mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DefaultStatementAttributes.java

```diff
@@ -0,0 +1,183 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.mapper.StatementAttributes;
+import com.datastax.oss.driver.api.mapper.StatementAttributesBuilder;
+import com.datastax.oss.protocol.internal.util.collection.NullAllowingImmutableMap;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Map;
+
+public class DefaultStatementAttributes implements StatementAttributes {
+
+  private DriverExecutionProfile profile = null;
+
+  private String profileName = null;
+
+  private Boolean isIdempotent = null;
+
+  private int pageSize = -Integer.MIN_VALUE;
+
+  private ConsistencyLevel consitencyLevel = null;
+
+  private ConsistencyLevel serialConsistencyLevel = null;
+
+  private Duration timeout = null;
+
+  private ByteBuffer routingKey = null;
+
+  private String routingKeyspace = null;
+
+  private boolean tracing = false;
+
+  private ByteBuffer pagingState = null;
+
+  private long timeStamp = Long.MIN_VALUE;
+
+  private Node node = null;
+
+  private Map<String, ByteBuffer> customPayload = NullAllowingImmutableMap.of();
+
+  public static StatementAttributesBuilder builder() {
+    return new StatementAttributesBuilder();
+  }
+
+  @Override
+  public DriverExecutionProfile getExecutionProfile() {
+    return profile;
+  }
+
+  @Override
+  public String getExecutionProfileName() {
+    return profileName;
+  }
+
+  @Override
+  public Boolean isIdempotent() {
+    return isIdempotent;
+  }
+
+  @Override
+  public int getPageSize() {
+    return pageSize;
+  }
+
+  @Override
+  public ConsistencyLevel getConsistencyLevel() {
+    return consitencyLevel;
+  }
+
+  @Override
+  public ConsistencyLevel getSerialConsistencyLevel() {
+    return serialConsistencyLevel;
+  }
+
+  @Override
+  public Duration getTimeout() {
+    return timeout;
+  }
+
+  @Override
+  public ByteBuffer getRoutingKey() {
+    return routingKey;
+  }
+
+  @Override
+  public String getRoutingKeyspace() {
+    return routingKeyspace;
+  }
+
+  @Override
+  public boolean isTracing() {
+    return tracing;
+  }
+
+  @Override
+  public ByteBuffer getPagingState() {
+    return pagingState;
+  }
+
+  @Override
+  public long getTimestamp() {
+    return timeStamp;
+  }
+
+  @Override
+  public Node getNode() {
+    return node;
+  }
+
+  @Override
+  public Map<String, ByteBuffer> getCustomPayload() {
+    return customPayload;
+  }
+
+  public void setProfile(DriverExecutionProfile profile) {
+    this.profile = profile;
+  }
```

Comment:
Could you make this class immutable instead?
* final fields
* a constructor that takes all the fields
* no setters
* the builder has its own fields, and calls the constructor in `build()`.

See `SimpleStatement` in the core driver, it follows a similar pattern.

The problem with mutability is that it's not thread-safe.